homeassistant/components/fully_kiosk/entity.py

```diff
@@ -17,12 +17,16 @@ class FullyKioskEntity(CoordinatorEntity[FullyKioskDataUpdateCoordinator], Entit
     def __init__(self, coordinator: FullyKioskDataUpdateCoordinator) -> None:
         """Initialize the Fully Kiosk Browser entity."""
         super().__init__(coordinator=coordinator)
-        self._attr_device_info = DeviceInfo(
-            identifiers={(DOMAIN, coordinator.data["deviceID"])},
-            name=coordinator.data["deviceName"],
-            manufacturer=coordinator.data["deviceManufacturer"],
-            model=coordinator.data["deviceModel"],
-            sw_version=coordinator.data["appVersionName"],
-            configuration_url=f"http://{coordinator.data['ip4']}:2323",
-            connections={(CONNECTION_NETWORK_MAC, coordinator.data["Mac"])},
-        )
+        device_params = {
+            "identifiers": {(DOMAIN, coordinator.data["deviceID"])},
+            "name": coordinator.data["deviceName"],
+            "manufacturer": coordinator.data["deviceManufacturer"],
+            "model": coordinator.data["deviceModel"],
+            "sw_version": coordinator.data["appVersionName"],
+            "configuration_url": f"http://{coordinator.data['ip4']}:2323",
+        }
+        if "Mac" in coordinator.data and coordinator.data["Mac"]:
+            device_params["connections"] = {
+                (CONNECTION_NETWORK_MAC, coordinator.data["Mac"])
+            }
+        self._attr_device_info = DeviceInfo(**device_params)  # type: ignore[misc]
```

Comment:
How about something like this instead so we don't have to work against the type checker:
```suggestion
        device_info = DeviceInfo(
            identifiers={(DOMAIN, coordinator.data["deviceID"])},
            name=coordinator.data["deviceName"],
            manufacturer=coordinator.data["deviceManufacturer"],
            model=coordinator.data["deviceModel"],
            sw_version=coordinator.data["appVersionName"],
            configuration_url=f"http://{coordinator.data['ip4']}:2323",
        )
        if "Mac" in coordinator.data and coordinator.data["Mac"]:
            device_info["connections"] = {
                (CONNECTION_NETWORK_MAC, coordinator.data["Mac"])
            }
        self._attr_device_info = device_info
```